bridge/src/commonMain/kotlin/ru/herobrine1st/matrix/bridge/telegram/TelegramWorker.kt

```diff
@@ -50,97 +52,138 @@ class TelegramWorker(
-                            is RoomMessageEventContent.TextBased.Text -> {
-                                val textContent = content.body
-
-                                withContext(Dispatchers.IO) {
-                                    bot.sendMessage(
-                                        chatId = roomId,
-                                        text = "<${event.sender}>: $textContent",
-                                    )
-                                }
-                            }
+                val contentRaw = event.content
+                if (contentRaw !is RoomMessageEventContent) return
 
-                            is RoomMessageEventContent.FileBased.Image -> {
-                                val caption = when {
-                                    content.fileName == null || content.fileName == content.body ->
-                                        "${event.sender} sent a picture"
+                val relatesTo = contentRaw.relatesTo
 
-                                    else -> "${event.sender}: ${content.body}"
-                                }
+                if (relatesTo?.relationType == RelationType.Thread) {
+                    return
+                }
 
-                                // encrypted or malformed if null
-                                val url = content.url ?: return
-
-                                downloadMatrixMedia(url) { photoFile ->
-                                    withContext(Dispatchers.IO) {
-                                        bot.sendPhoto(
-                                            chatId = roomId,
-                                            photo = photoFile,
-                                            caption = caption,
-                                        )
-                                    }.toResult()
-                                }
-                            }
+                val editedTelegramMessageId = when (relatesTo) {
+                    is RelatesTo.Replace -> api.getMessageEventId(relatesTo.eventId)
+                    else -> null
+                }
 
-                            is RoomMessageEventContent.FileBased.File -> {
-                                val caption = when {
-                                    content.fileName == null || content.fileName == content.body ->
-                                        "${event.sender} sent a file"
+                val content = if (editedTelegramMessageId != null && relatesTo is RelatesTo.Replace) {
+                    relatesTo.newContent
+                } else {
+                    contentRaw
+                }
 
-                                    else -> "${event.sender}: ${content.body}"
-                                }
+                if (content !is RoomMessageEventContent) {
+                    throw UnhandledEventException(
+                        "Got something other than RoomMessageEventContent from replacement event",
```

Comment:
Or good luck debugging that

```suggestion
                        "Got something other than RoomMessageEventContent from replacement event: $content",
```